packages/match/controller/src/index.js

```diff
@@ -19,26 +20,30 @@ const getResponseCorrectness = (
     };
   }
 
+  if (rows.length) {
+    totalCorrectAnswers += rows.length * (model.layout - 1);
+  }
+
   rows.forEach(row => {
-    const isCorrectAnswer = answers[row.id] && JSON.stringify(answers[row.id]) === JSON.stringify(row.values);
+    const answer = answers[row.id];
 
-    if (isCorrectAnswer) {
-      correctAnswers += 1;
+    if (answer) {
+      row.values.forEach((v, i) => {
+        if (answer[i] === v) {
+          correctAnswers += 1;
+        }
+      });
     }
   });
 
-  if (rows.length === correctAnswers) {
+  if (totalCorrectAnswers === correctAnswers) {
     return { correctness: 'correct', score: '100%' };
   } else if (correctAnswers === 0) {
     return { correctness: 'incorrect', score: '0%' };
-  } else if (allowPartialScores && partialScores && partialScores.length) {
+  } else if (allowPartialScores && partialScoring) {
```

Comment:
scoring logic needs to be in the `outcome` method, if the ui wants it too in evaluate mode it can call `outcome`, note that outcome should only return a value between 0.0 - 1.0, so if the ui wants a string % you can convert it in the model fn.